main/util/src/mill/util/Version.scala

```diff
@@ -0,0 +1,161 @@
+package mill.util
+
+import mill.api.experimental
+
+class Version private (
```

Comment:
I wonder how many versions of something like this are floating around the ecosystem. For example we have one in Metals as well https://github.com/scalameta/metals/blob/main/mtags/src/main/scala/scala/meta/internal/semver/SemVer.scala 😆.

Reply:
Yeah, the issue is, that such simple yet non-trivial helper neither deserves its own library nor its worth to pull in some larger existing library into the classpath.